packages/xstate-graph/src/shortestPaths.ts

```diff
@@ -133,8 +132,40 @@ export function getShortestPaths<
   );
 
   if (resolvedOptions.toState) {
-    return paths.filter((path) => resolvedOptions.toState!(path.state));
+    return paths
+      .filter((path) => resolvedOptions.toState!(path.state))
+      .map(alterPath);
   }
 
-  return paths;
+  return paths.map(alterPath);
+}
+
+export function alterPath<T extends StatePath<any, any>>(path: T): T {
```

Comment:
I think that this should be moved to a different file since this is used by both `shortestPaths` and `simplestPaths`